scripts/components/bookmarks/index.es6

```diff
@@ -0,0 +1,144 @@
+import _ from 'underscore';
+import _str from 'underscore.string';
+import api from 'api';
+import BookmarksListItem from 'components/bookmarks/list-item';
+import Cursors from 'cursors';
+import Empty from 'components/shared/empty';
+import ErrorBlock from 'components/ui/error-block';
+import Filters from 'components/bookmarks/filters';
+import List from 'react-list';
+import LoadingBlock from 'components/ui/loading-block';
+import React from 'react';
+
+var PER_PAGE = 10;
+
+export default React.createClass({
+  mixins: [Cursors],
+
+  propTypes: {
+    /* Specify which portal's bookmarks to retrieve */
+    portalId: React.PropTypes.number.isRequired,
+  },
+
+  getDefaultProps: function () {
+    return {
+      bookmarks: [],
+      filtersAreShowing: true,
+      query: '',
+      searchableAttributes: ['name']
+    };
+  },
+
+  getInitialState: function () {
+    return {
+      bookmarks: this.props.bookmarks,
+      query: this.props.query
+    };
+  },
+
+  fetch: function (cb) {
+    api.get('/portals/:portal_id/links', {
+      portal_id: this.props.portalId,
+      page: Math.floor(this.state.bookmarks.length / PER_PAGE) + 1,
+      per_page: PER_PAGE
+    }, _.partial(this.handleFetch, cb));
+  },
+
+  handleFetch: function (cb, er, res) {
+    if (er) return cb(er);
+    this.update({
+      bookmarks: {$set: _.unique(this.state.bookmarks.concat(res.data), 'id')}
+    });
+    cb(null, res.data.length < PER_PAGE);
+  },
+
+  matchesQuery: function(bookmark) {
+    var query = this.state.query;
+    if (!query) return true;
+    var words = _str.words(query.toLowerCase());
+    var searchableWords = this.searchableWordsFor(bookmark);
+    return _.every(words, function (wordA) {
+      return _.any(searchableWords, function (wordB) {
+        return _str.startsWith(wordB, wordA);
+      });
+    });
+  },
+
+  searchableWordsFor: function (bookmark) {
+    return _str.words(
+      _.values(_.pick(bookmark, this.props.searchableAttributes))
+        .join(' ')
+        .toLowerCase()
+    );
+  },
+
+  bookmarkMatchesFilters: function(bookmark) {
+    return this.matchesQuery(bookmark);
+  },
+
+  getFilteredBookmarks: function() {
+    return this.state.bookmarks.filter(this.bookmarkMatchesFilters);
+  },
+
+  renderFilters: function (bookmarks) {
+    if (!this.state.bookmarks.length || !this.props.filtersAreShowing) return;
+    return (
+      <Filters
+        bookmarks={bookmarks}
+        getFacet={this.getFacet}
+        cursors={{
+          query: this.getCursor('query')
+        }}
+      />
+    );
+  },
+
+  renderListItem: function (bookmark) {
+    var i = this.state.bookmarks.indexOf(bookmark);
+    return (
+      <BookmarksListItem
+        {...this.props}
+        key={bookmark.id}
+        cursors={{bookmark: this.getCursor('bookmarks', i)}}
```

Comment:
`jshint` used to have a setting but I think it's been removed. A lot of people have been moving to `eslint` which does have this option (http://eslint.org/docs/rules/space-in-brackets). I'll switch us over if it looks good.